lib/Models/SdmxJson/SdmxJsonDataflowStratum.ts

```diff
@@ -0,0 +1,424 @@
+import { computed } from "mobx";
+import isDefined from "../../Core/isDefined";
+import TerriaError from "../../Core/TerriaError";
+import SdmxCatalogItemTraits from "../../Traits/SdmxCatalogItemTraits";
+import TableColumnTraits from "../../Traits/TableColumnTraits";
+import createStratumInstance from "../createStratumInstance";
+import LoadableStratum from "../LoadableStratum";
+import { BaseModel } from "../Model";
+import proxyCatalogItemUrl from "../proxyCatalogItemUrl";
+import StratumOrder from "../StratumOrder";
+import SdmxJsonCatalogItem from "./SdmxJsonCatalogItem";
+import { loadSdmxJsonStructure, parseSdmxUrn } from "./SdmxJsonServerStratum";
+import {
+  CodeLists,
+  ConceptScheme,
+  ConceptSchemes,
+  ContentConstraints,
+  Dataflow,
+  DataStructure,
+  SdmxJsonStructureMessage
+} from "./SdmxJsonStructureMessage";
+
+export interface SdmxJsonDataflow {
+  dataflow: Dataflow;
+  dataStructure: DataStructure;
+  codelists?: CodeLists;
+  conceptSchemes?: ConceptSchemes;
+  contentConstraints?: ContentConstraints;
+}
+export class SdmxJsonDataflowStratum extends LoadableStratum(
+  SdmxCatalogItemTraits
+) {
+  static stratumName = "sdmxJsonDataflow";
+
+  duplicateLoadableStratum(model: BaseModel): this {
+    return new SdmxJsonDataflowStratum(
+      model as SdmxJsonCatalogItem,
+      this.sdmxJsonDataflow
+    ) as this;
+  }
+
+  static async load(
+    catalogItem: SdmxJsonCatalogItem
+  ): Promise<SdmxJsonDataflowStratum> {
+    // Load dataflow (+ all related references)
+    let dataflowStructure: SdmxJsonStructureMessage = await loadSdmxJsonStructure(
+      proxyCatalogItemUrl(
+        catalogItem,
+        `${catalogItem.url}/dataflow/${catalogItem.agencyId}/${catalogItem.dataflowId}?references=all`
+      ),
+      false
+    );
+
+    // Check response
+    if (!isDefined(dataflowStructure.data)) {
+      throw new TerriaError({
+        title: `Could not load SDMX Dataflow`,
```

Comment:
I know it's painful but do we want to internationalize these strings?

Reply:
yep, no problem